### ISSUE TITLE
/home/{user}/.fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o

### ISSUE THREAD

**Issue by fluency03** [junegunn/fzf#2094]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish
    
## Problem / Steps to reproduce

I have followed the instruction to install fzf using git way.

I have selected to use key binding.

After installation and `source ~/.zshrc`, I have this error:

```
/home/{my-user-name}/.fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o
```

And it keeps popping up every time I login to the host.

**Comment by junegunn** [comment]
`zsh --version`?

**Comment by fluency03** [comment]
zsh 5.7.1 (x86_64-pc-linux-gnu)

**Comment by junegunn** [comment]
I can't reproduce the problem, tested on zsh 5.4 and 5.8

```sh
$ echo $ZSH_VERSION
5.4.2
$ source ~/.fzf.zsh
$ bindkey | grep fzf
"^I" fzf-completion
"^R" fzf-history-widget
"^T" fzf-file-widget
"^[c" fzf-cd-widget
```

**Comment by fluency03** [comment]
I have this correctly:

```
~ ➜ bindkey | grep fzf
"^I" fzf-completion
"^R" fzf-history-widget
"^T" fzf-file-widget
"^[c" fzf-cd-widget
```

But I just keeps getting the error when logon to the host:
```
/home/{user}/.fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o
```

**Comment by rohanghige** [comment]
I too see the same error whenever I do login:
`/home/$USER/fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o`

**Comment by shivshnkr** [comment]
+1 . I see this error too. Any workarounds ?

**Comment by pplmx** [comment]
Same issue.
`zsh --version`: `zsh 4.3.11 (x86_64-redhat-linux-gnu)`
fzf is the latest.

**Comment by izxle** [comment]
> Same issue. `zsh --version`: `zsh 4.3.11 (x86_64-redhat-linux-gnu)` fzf is the latest.
> 
> $ cat /etc/centos-release CentOS release 6.9 (Final)
> 
> ![image](https://user-images.githubusercontent.com/26994103/191178999-5a027ae3-3187-4412-bea5-d226b365bf19.png)

I have the same problem withe the exact same output.

**Comment by LangLangBart** [comment]
> I can't reproduce the problem, tested on zsh 5.4 and 5.8

Confirmed, with the default `zsh` version or the latest `brew` version the issue is not reproducible on macOS.


---

<details>
<summary>Building zsh from source</summary>

- instructions: https://sourceforge.net/p/zsh/code/ci/master/tree/INSTALL

1. clone
```sh
git clone https://git.code.sf.net/p/zsh/code zsh-code 
cd zsh-code 
```

2. pre-configure
> [!IMPORTANT]
> If the `configure` script does not already exist -- e.g., if you've got
a snapshot of the bare sources just checked out from a git repository
-- some things need to be built before the configuration can proceed.
Run the script `./Util/preconfig` to do this.

```sh
./Util/preconfig
```

3.  Configure
- define the location where the binary shall be placed 
```sh
./configure --prefix=$HOME/.local/zsh-test
```

4. check out the desired commit/tag
```sh
git checkout zsh-4.3.17 
```

5. Make the binary
```sh
make install.bin
```

6. Optionally add the binary to your PATH
```sh
# .zshrc
export PATH="$HOME/.local/zsh-test/bin:$PATH"
```

5. Run the binary
```sh
zsh --version
# zsh 4.3.17 (x86_64-apple-darwin19.6.0)
```

</details>

---

I can produce the error by running the following command with zsh version `4.3.17` non-interactively.

```sh
zsh --version
# zsh 4.3.17-dev-0 (x86_64-apple-darwin19.6.0)

# prints the current options flags set for that shell session
zsh -c 'echo $-'
# 569Xd

zsh -c 'source /Users/paria/Developer/fzf/shell/key-bindings.zsh'
# /Users/paria/Developer/fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o
```

```sh
❯ git tag --sort="-creatordate" --format="%(creatordate:format:%d/%b/%y) - %(refname:short)" | grep -C 2 'zsh-4.3.17'
# 16/Dec/12 - zsh-5.0.0-test-1
# 21/Jul/12 - zsh-5.0.0
# 15/Jul/12 - zsh-4.3.17-test-2
# 23/Feb/12 - zsh-4.3.17
# 20/Feb/12 - zsh-4.3.16
# 17/Dec/11 - zsh-4.3.15
```

When bisecting the `zsh` source code the issue was fixed with [zsh / Code / Commit [fa8a0e]](https://sourceforge.net/p/zsh/code/ci/fa8a0e241c8ac9eb48ab9fc2d8bb3105088e0db9/) (7/Mar/12)

```sh
# Commit [fa8a0e]
git checkout fa8a0e >/dev/null
make install.bin >/dev/null
zsh -c 'source ~/Developer/fzf/shell/key-bindings.zsh'

# Commit prior to [fa8a0e]
git checkout fa8a0e^ >/dev/null
make install.bin >/dev/null
zsh -c 'source /Users/paria/Developer/fzf/shell/key-bindings.zsh'
# /Users/paria/Developer/fzf/shell/key-bindings.zsh:emulate:35: unknown argument -o
```

---

- the line was added with #1944
  - merged in b1b916ce157df4fe1299edfdd012676419e2263c 
```sh
'emulate' 'zsh' '-o' 'no_aliases'
```
- @romkatv can you help here ?

---
### workaround 

- if you use the version prior to the merged commit, I assume you will not have the issue
- [History for shell/key-bindings.zsh - junegunn/fzf](https://github.com/junegunn/fzf/commits/b1b916ce157df4fe1299edfdd012676419e2263c/shell/key-bindings.zsh)

```sh
wget -qO key-zsh-old.sh https://raw.githubusercontent.com/junegunn/fzf/18261fe31cf1270f9aa783f99bbdbca343f89479/shell/key-bindings.zsh

zsh --version
zsh -c 'source key-zsh-old.sh '
```

**Comment by LangLangBart** [comment]
possible fix, check if interactive ? 😬 

```diff
--- a/shell/key-bindings.zsh
+++ b/shell/key-bindings.zsh
@@ -32,7 +32,7 @@ else
   }
 fi
 
-'emulate' 'zsh' '-o' 'no_aliases'
+[[ -o interactive ]] && 'emulate' 'zsh' '-o' 'no_aliases'
 
 {
 ```

**Comment by romkatv** [comment]
> * the line was added with [[zsh] Ensure that fzf code always parses the same way #1944](https://github.com/junegunn/fzf/pull/1944)
>   
>   * merged in [b1b916c](https://github.com/junegunn/fzf/commit/b1b916ce157df4fe1299edfdd012676419e2263c)
> 
> ```shell
> 'emulate' 'zsh' '-o' 'no_aliases'
> ```
> 
> * @romkatv can you help here ?

The `-o` flag was added to `emulate` in zsh 5.0.2. If you want to support zsh older than that, you'll have to avoid this flag and use `setopt` to set options.

```zsh
'builtin' 'emulate' 'zsh' && 'builtin' 'setopt' 'no_aliases'
```
